src/CMakeLists.txt

```diff
@@ -366,6 +366,7 @@ add_library(Halide
             $<TARGET_OBJECTS:Halide_initmod>)
 add_library(Halide::Halide ALIAS Halide)
 
+target_link_libraries(Halide PRIVATE Threads::Threads)
```

Comment:
If any of our headers include `<thread>`, then this should be `PUBLIC`. Otherwise, if the only header that includes it doesn't need it, then the include should be dropped and this can remain `PRIVATE`.

Reply:
Ah, i see why i couldn't find any includes of the `threadpool.h`, because `halide.h` is an autogenerated amalgamation :/
If i remove that header, building of tests starts failing.